src/main/java/org/jenkinsci/plugins/casc/VaultSecretSource.java

```diff
@@ -29,7 +28,7 @@ public String reveal(String fromKey) {
         String vaultUrl = System.getenv("CASC_VAULT_URL");
         if(vaultPw != null && vaultUsr != null && vaultPth != null && vaultUrl != null) {
             try {
-                Matcher m = VAULT_VARIABLE.matcher(fromKey);
+                Matcher m = SecretSource.SECRET_PATTERN.matcher(fromKey);
                 if (m.matches()) {
                     final String vaultKey = m.group(1);
                     VaultConfig config = new VaultConfig().address(vaultUrl).build();
```

Comment:
move this to a constructor so this isn't ran for each and every variable subsitution

Reply:
I can't really move the matcher to a constructor since it requires a key to construct which we parse from the yml? What about making a static method on SecretSource like `Optional<String> requiresReveal(String key)` which would create the matcher, and return the captured content from the regex once, for every key that needs to be revealed? Then i can remove the matcher from both SecretSources?

Reply:
You can move this whole block into constructor
```java
String vaultPw = System.getenv("CASC_VAULT_PW");
String vaultUsr = System.getenv("CASC_VAULT_USER");
String vaultPth = System.getenv("CASC_VAULT_PATH");
String vaultUrl = System.getenv("CASC_VAULT_URL");

if (vaultPw != null && vaultUsr != null && vaultPth != null && vaultUrl != null) {
try {
    VaultConfig config = new VaultConfig().address(vaultUrl).build();
    Vault vault = new Vault(config);
    // Obtain a login token
    final String token = vault.auth().loginByUserPass(vaultUsr, vaultPw).getAuthClientToken();
    config.token(token).build();
    // Fetch
    Map secrets = vault.logical().read(vaultPth).getData()
```
can't you ? Then make `secrets` an attribute, and limit `reveal` to a key lookup in this Map.

Reply:
@ndeloof that i can do :+1: